src/site/markdown/server/manage-webadmin.md

```diff
@@ -2496,6 +2497,45 @@ Response codes:
 
 Not implemented yet.
 
+## Deleted Messages Vault
+
+Deleted messages from an user will be stored in a restriction place where only delegated administrator can access.
+This place is called the Vault. Messages in the Vault will be deleted permanently by a configured retention policy.
+
+ - [Restore Deleted Messages](#Restore_deleted_messages)
+
+### Restore Deleted Messages
+
+An user can request to the administrator to restore his messages had been deleted and now at the Vault.
+To restore them, the administrator perform a http request:
+
+```
+curl -XPOST http://ip:port/deletedMessages/user/userToRestore@domain.ext
+```
+
+**All** messages from the user passed in the path param will be moved from the Vault to his new mailbox.
+Response code:
+
+ - 201: Task for restoring deleted has been created
+ - 400: Bad request, user parameter is invalid
+
+The scheduled task will have the following type `deletedMessages/restore` and the following `additionalInformation`:
+
+```
+{
+  "successfulRestoreCount": 47,
+  "errorRestoreCount": 0
+  "failRemovingCount": 19
+  "user": "userToRestore@domain.ext"
+}
+```
+
+while:
+ - successfulRestoreCount: number of messages has been appended to user restore mailbox
+ - errorRestoreCount: number of messages has **not** been appended to user restore mailbox
+ - failRemovingCount: number of messages failed to delete in the Vault after storing corresponding message to user mailbox
```

Comment:
I'm not sure why we even try to delete these messages. Could you explain that choice?

Reply:
they are the messages expected to be removed from the vault after appended to user mailbox. but maybe by some how vault.delete() gets errors, and we decided to not rollback the restore for that message - just keep message in user mailbox rather than delete it if delete in the vault fails. I want to log the number of that messages. 

Converstation is here: https://github.com/linagora/james-project/pull/2197#discussion_r260576167

Reply:
I read https://github.com/linagora/james-project/pull/2197#discussion_r260576167 but my question is broader than that: why would we remove the message from vault at all?
I don't see how it make the feature better.

Reply:
How about two cases:
- If I restore my deleted mails two times, and I will see in my mailbox messages those are duplicated two times.
- If I restore my deleted mails, then I delete them, the vault will store duplicated messages, actually they are the same messages.